test/integration-tests/shared/protocolHelpers/permit2.ts

```diff
@@ -72,48 +80,96 @@ export function getEip712Domain(chainId: number, verifyingContract: string) {
   }
 }
 
-export async function signPermit(
-  permit: PermitSingle,
-  signer: SignerWithAddress,
-  verifyingContract: string
-): Promise<string> {
-  const eip712Domain = getEip712Domain(chainId, verifyingContract)
-  const signature = await signer._signTypedData(eip712Domain, PERMIT2_PERMIT_TYPE, permit)
-
-  return signature
-}
-
 export async function getPermitSignature(
-  permit: PermitSingle,
-  signer: SignerWithAddress,
+  permitSingle: PermitSingle,
+  signer: ethers.utils.SigningKey,
   permit2: Permit2
 ): Promise<string> {
   // look up the correct nonce for this permit
-  const nextNonce = (await permit2.allowance(signer.address, permit.details.token, permit.spender)).nonce
-  permit.details.nonce = nextNonce
-  return await signPermit(permit, signer, permit2.address)
+  const signerAddress = ethers.utils.computeAddress(signer.privateKey)
+  const nextNonce = (await permit2.allowance(signerAddress, permitSingle.details.token, permitSingle.spender)).nonce;
+  permitSingle.details.nonce = nextNonce;
+  const provider = Provider.getDefaultProvider();
+  const network = await provider.getNetwork();
+  const chainId = network.chainId;
+  const verifyingContract = permit2.address;
+  const DOMAIN_SEPARATOR = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(
+      ['bytes32', 'bytes32', 'uint256', 'address'],
+      [_TYPE_HASH, _HASHED_NAME, chainId, verifyingContract]
+    )
+  );
+  const permitDetailsHash = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(
+      ['bytes32', 'address', 'uint160', 'uint48', 'uint48'],
+      [_PERMIT_DETAILS_TYPEHASH, permitSingle.details.token, permitSingle.details.amount, permitSingle.details.expiration, permitSingle.details.nonce]
+    )
+  );
+  const permitSingleHash = ethers.utils.keccak256(
+    ethers.utils.defaultAbiCoder.encode(
+      ['bytes32', 'bytes32', 'address', 'uint256'],
+      [_PERMIT_SINGLE_TYPEHASH, permitDetailsHash, permitSingle.spender, permitSingle.sigDeadline]
+    )
+  );
+  const hashTypedData = ethers.utils.keccak256(
+    ethers.utils.hexConcat([
+      ethers.utils.arrayify(ethers.utils.toUtf8Bytes('\x19\x01')),
+      DOMAIN_SEPARATOR,
+      permitSingleHash
+    ])
+  );
+  const signature: Signature = signer.signDigest(hashTypedData)
+  return signature.compact
 }
 
 export async function getPermitBatchSignature(
-  permit: PermitBatch,
-  signer: SignerWithAddress,
+  permitBatch: PermitBatch,
+  signer: ethers.utils.SigningKey,
   permit2: Permit2
 ): Promise<string> {
-  for (const i in permit.details) {
-    const nextNonce = (await permit2.allowance(signer.address, permit.details[i].token, permit.spender)).nonce
-    permit.details[i].nonce = nextNonce
+  for (const i in permitBatch.details) {
```

Comment:
Why it's implemented manually?

Reply:
I dont remember exactly what problem we had, but  we solved it by implementing manually.

Reply:
I think zksync wallet should support signTypedData. If there are no concerns, can you please use it?